src/configuration-requestor.ts

```diff
@@ -1,150 +1,121 @@
+
+export class ConfigurationError extends Error {
+  public constructor(message: string) {
+    super(message);
+    this.name = 'ConfigurationError';
+  }
+}
+
+/** @internal */
+export type ConfigurationRequestorOptions = {
+  wantsBandits: boolean;
+
+  precomputed?: {
+    subjectKey: string;
+    subjectAttributes: ContextAttributes;
+    banditActions?: Record<FlagKey, Record<string, ContextAttributes>>;
+  };
+};
+
+/**
+ * @internal
+ */
 export default class ConfigurationRequestor {
-  private banditModelVersions: string[] = [];
-  private configuration: StoreBackedConfiguration;
+  private readonly options: ConfigurationRequestorOptions;
 
-  constructor(
+  // We track the latest seen configuration to possibly reuse it for flags/bandits.
+  private latestConfiguration?: Configuration;
+
+  public constructor(
     private readonly httpClient: IHttpClient,
-    private flagConfigurationStore: IConfigurationStore<Flag>,
-    private banditVariationConfigurationStore: IConfigurationStore<BanditVariation[]> | null,
-    private banditModelConfigurationStore: IConfigurationStore<BanditParameters> | null,
+    private readonly configurationFeed: ConfigurationFeed,
+    options: Partial<ConfigurationRequestorOptions> = {},
   ) {
-    this.configuration = new StoreBackedConfiguration(
-      this.flagConfigurationStore,
-      this.banditVariationConfigurationStore,
-      this.banditModelConfigurationStore,
-    );
-  }
+    this.options = {
+      wantsBandits: true,
+      ...options,
+    };
 
-  /**
-   * Updates the configuration stores and recreates the StoreBackedConfiguration
-   */
-  public setConfigurationStores(
-    flagConfigurationStore: IConfigurationStore<Flag>,
-    banditVariationConfigurationStore: IConfigurationStore<BanditVariation[]> | null,
-    banditModelConfigurationStore: IConfigurationStore<BanditParameters> | null,
-  ): void {
-    this.flagConfigurationStore = flagConfigurationStore;
-    this.banditVariationConfigurationStore = banditVariationConfigurationStore;
-    this.banditModelConfigurationStore = banditModelConfigurationStore;
-
-    // Recreate the configuration with the new stores
-    this.configuration = new StoreBackedConfiguration(
-      this.flagConfigurationStore,
-      this.banditVariationConfigurationStore,
-      this.banditModelConfigurationStore,
-    );
-  }
+    this.configurationFeed.addListener((configuration) => {
+      const prevFetchedAt = this.latestConfiguration?.getFetchedAt();
+      const newFetchedAt = configuration.getFetchedAt();
 
-  public isFlagConfigExpired(): Promise<boolean> {
-    return this.flagConfigurationStore.isExpired();
+      if (!prevFetchedAt || (newFetchedAt && newFetchedAt > prevFetchedAt)) {
+        this.latestConfiguration = configuration;
+      }
+    });
   }
 
-  public getConfiguration(): IConfiguration {
-    return this.configuration;
+  public async fetchConfiguration(): Promise<Configuration> {
```

Comment:
`fetchConfiguration()` now returns full `Configuration` object

ConfigurationRequestor is now also able to handle both regular and precomputed requests